### ISSUE TITLE
Enhancement: system check command

### ISSUE THREAD

**Issue by robbieaverill** [silverleague/silverstripe-console#47]
A `system:check` command could perform a series of automated checks on your environment. This wouldn't necessarily replace things like the silverstripe/environmentcheck module, but more basic checks that would be required for SilverStripe to operate (NB: it's possible that the command itself may not run if that were the case):

* PHP version support
* php-intl installed
* dotenv is available
* Permissions across various folders (read everywhere, local write on assets)
* Database is reachable

**Comment by robingram** [comment]
Is there any gain from having this outside the installation process? I'm assuming that those things are checked on install, or should be.

I guess one use would be troubleshooting problems after deploying to a new server.

**Comment by robbieaverill** [comment]
@robingram yeah they're checked on install provided you use the web installer. If you configure it yourself then you will have to rely on any composer requirements to check your server specs.

**Comment by robbieaverill** [comment]
Perhaps this might be more useful as a `diagnose` command, doing something similar to what `composer diagnose` does - system information, SilverStripe version information etc
